ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/BlockAttestationsPoolTest.java

```diff
@@ -13,44 +13,159 @@
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes32;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.datastructures.operations.AttestationData;
+import tech.pegasys.artemis.datastructures.util.DataStructureUtil;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.util.SSZTypes.Bitlist;
+import tech.pegasys.artemis.util.bls.BLSKeyGenerator;
+import tech.pegasys.artemis.util.bls.BLSKeyPair;
 
 class BlockAttestationsPoolTest {
 
+  private final List<BLSKeyPair> validatorKeys = BLSKeyGenerator.generateKeyPairs(12);
+  private final ChainStorageClient storageClient =
+      spy(new ChainStorageClient(mock(EventBus.class)));
+  private final AttestationGenerator attestationGenerator = new AttestationGenerator(validatorKeys);
   private BlockAttestationsPool pool;
 
   @BeforeEach
   void setup() {
-    pool = spy(new BlockAttestationsPool());
+    BeaconChainUtil.initializeStorage(storageClient, validatorKeys);
+    pool = new BlockAttestationsPool();
   }
 
   @Test
-  void unprocessedAggregate_SingleAttester_ShouldBeIgnored() throws Exception {}
+  void unprocessedAggregate_SingleAttester_ShouldBeIgnored() throws Exception {
+    final Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    pool.addUnprocessedAggregateAttestationToQueue(attestation);
+    assertFalse(pool.aggregateAttesationsQueue.contains(attestation));
+    assertFalse(
+        pool.unprocessedAttestationsBitlist.containsValue(attestation.getAggregation_bits()));
+  }
 
   @Test
-  void unprocessedAggregate_NewData_BitlistAdded() throws Exception {}
+  @SuppressWarnings("UnusedVariable")
+  void unprocessedAggregate_NewData() throws Exception {
+    Attestation attestation = AttestationGenerator.aggregateAttestation(10);
+    pool.addUnprocessedAggregateAttestationToQueue(attestation);
+    assertTrue(pool.aggregateAttesationsQueue.contains(attestation));
+    assertTrue(
+        pool.unprocessedAttestationsBitlist.containsValue(attestation.getAggregation_bits()));
+  }
 
   @Test
-  void unprocessedAggregate_OldData_DifferentBitlist_BitlistUpdated() throws Exception {}
+  void unprocessedAggregate_OldData_DifferentBitlist_BitlistUpdated() throws Exception {
+    Attestation attestation = AttestationGenerator.aggregateAttestation(10);
+    Attestation newAttestation = withNewAttesterBits(new Attestation(attestation), 1);
+
+    pool.addUnprocessedAggregateAttestationToQueue(attestation);
+    pool.addUnprocessedAggregateAttestationToQueue(newAttestation);
+    Bytes32 attestationDataHash = attestation.getData().hash_tree_root();
+    Bitlist bitlist = pool.unprocessedAttestationsBitlist.get(attestationDataHash);
+    for (int i = 0; i < attestation.getAggregation_bits().getCurrentSize(); i++) {
+      if (bitlist.getBit(i) != attestation.getAggregation_bits().getBit(i)
```

Comment:
Not sure this check is right,  you could simplify to something like:
```
final int expected = attestation.getAggregation_bits().getBit(i) | newAttestation.getAggregation_bits().getBit(i);
assertThat(bitlist.getBit(i)).isEqualTo(expected);
```

Reply:
done